### PR TITLE
FEAT: Added getitem to primitives. fixed bug in getitem for padstack. 

### PR DESCRIPTION
**Pull request by maxcapodi78**
Added cache data of instances and definitions properties to make it faster to retrieve info

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/layout.py
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/layout.py:151]: Is it possible that we end up adding an element to `self._layout.primitives` and the removing another one, making `self._layout.primitives` having the same length but with a different content ? If yes, then I think we should leverage this if condition as one might end up with incorrect cached data.

File: src/pyedb/dotnet/edb_core/padstack.py
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/padstack.py:188]: Same question as previous comment

File: src/pyedb/dotnet/edb_core/padstack.py
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/padstack.py:225]: Same question as first comment

File: src/pyedb/dotnet/edb_core/layout.py
Comment (maxcapodi78) [reply src/pyedb/dotnet/edb_core/layout.py:151]: I've added a cleanup of the variables on the delete operation so we should be safe enough.

**Review by ring630 — CHANGES_REQUESTED** [review]
@maxcapodi78 there is a method FindById which gets primitives from layout fast. getitem can benefit from it and gets much faster. It is implemented in another PR https://github.com/ansys/pyedb/pull/649. That PR is a complete refactor of EDBLayout. Please wait until it's merged.

**Review by svandenb-dev — APPROVED** [review]
LGTM

File: src/pyedb/dotnet/edb_core/cell/layout_obj.py
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/layout_obj.py:94]: Should we also do that when on updates `_edb_properties` since the value might be changed ? I'm wondering if this could lead to a cached object that does not contain the correct value this time.

Comment (maxcapodi78) [reply src/pyedb/dotnet/edb_core/cell/layout_obj.py:94]: everything in the object is not cached. So there is no need to refresh when you edit the property

**Review by ring630 — APPROVED** [review]

**Pull request merged**
